### PR TITLE
refactor(core): Don't genrate `standalone: true` for definitions

### PR DESCRIPTION
**Pull request by JeanMeche**
On top of #58236

See individual commits.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -85,8 +85,8 @@ jobs:
         run: yarn install --frozen-lockfile
       - name: Build adev in fast mode to ensure it continues to work
         run: yarn bazel build //adev:build --config=release
-      - name: Run tests
-        run: yarn bazel test //adev/...
+      # - name: Run tests
+      #   run: yarn bazel test //adev/...
```
Comment (devversion) [inline .github/workflows/ci.yml:89]: Leftover from debugging

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -114,8 +114,8 @@ jobs:
         run: yarn install --frozen-lockfile
       - name: Build adev in fast mode to ensure it continues to work
         run: yarn bazel build //adev:build --config=release
-      - name: Run tests
-        run: yarn bazel test //adev/...
+      # - name: Run tests
+      #   run: yarn bazel test //adev/...
```
Comment (devversion) [inline .github/workflows/pr.yml:118]: Same here

File: goldens/size-tracking/integration-payloads.json [hunk 3]
```diff
@@ -13,9 +13,9 @@
   },
   "cli-hello-world-lazy": {
     "uncompressed": {
-      "main": 108611,
-      "polyfills": 34169,
-      "lazy.routes": 388
+      "main": 107189,
+      "polyfills": 34544,
+      "lazy.routes": 368
```
Comment (devversion) [inline goldens/size-tracking/integration-payloads.json:18]: I think we should only update numbers that fell below I think, so we get notified if something moved us over the threshold over a longer period of time. Right now, we'd be hiding that if we regularly update numbers, regardless of "failures".

Did those fail?

File: packages/compiler-cli/test/compliance/test_cases/output_function/mixed_variants.ts [hunk 4]
```diff
@@ -2,7 +2,6 @@ import {Directive, EventEmitter, Output, output} from '@angular/core';
 import {outputFromObservable} from '@angular/core/rxjs-interop';
 
 @Directive({
-  standalone: true,
```
Comment (devversion) [inline packages/compiler-cli/test/compliance/test_cases/output_function/mixed_variants.ts:5]: Are these changes related? That seems like TS source code, and not "generated code"

File: packages/core/src/render3/definition.ts [hunk 5]
```diff
@@ -603,7 +604,7 @@ export function ɵɵdefinePipe<T>(pipeDef: {
     name: pipeDef.name,
     factory: null,
     pure: pipeDef.pure !== false,
-    standalone: pipeDef.standalone === true,
+    standalone: pipeDef.standalone === undefined ? NG_STANDALONE_DEFAULT_VALUE : pipeDef.standalone,
```
Comment (devversion) [inline packages/core/src/render3/definition.ts:607]:
```suggestion
    standalone: pipeDef.standalone ?? NG_STANDALONE_DEFAULT_VALUE,
```

File: packages/core/src/render3/definition.ts [hunk 6]
```diff
@@ -665,8 +665,10 @@ function getNgDirectiveDef<T>(directiveDefinition: DirectiveDefinition<T>): Dire
     inputTransforms: null,
     inputConfig: directiveDefinition.inputs || EMPTY_OBJ,
     exportAs: directiveDefinition.exportAs || null,
-    // TODO: standalone as default value (invert the condition)
-    standalone: directiveDefinition.standalone === true,
+    standalone:
```
Comment (devversion) [inline packages/core/src/render3/definition.ts:668]:
```suggestion
    standalone: directiveDefinition.standalone ?? NG_STANDALONE_DEFAULT_VALUE
```

**Review by devversion — APPROVED** [review]
LGTM, but two things:

- there is a typo in the commit message
- can we change the scope to `core` and mention that the runtime default is `standalone: true` now? I feel like this wasn't clear initially

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JeanMeche) [reply .github/workflows/ci.yml:89]: This is done on purpose, the ADEV test are broken because of the change, cf the dedicated commit. 
As we discussed it before, we'll be able to re-enable the tests once this is shipped and we can update the version ADEV is based on.

File: packages/compiler-cli/test/compliance/test_cases/output_function/mixed_variants.ts (hunk 4 above)
Comment (JeanMeche) [reply packages/compiler-cli/test/compliance/test_cases/output_function/mixed_variants.ts]: We can go without it.

File: goldens/size-tracking/integration-payloads.json (hunk 3 above)
Comment (JeanMeche) [reply goldens/size-tracking/integration-payloads.json:18]: Good catch, I assumed the "SUCCESS" were failures.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (devversion) [reply .github/workflows/ci.yml:89]: Ahh, I remember. We should have TODOs though, and link to a tracking issue

Comment (devversion) [reply .github/workflows/ci.yml:91]: @JeanMeche we can re-enable with a new FW release right?

Comment (JeanMeche) [reply .github/workflows/ci.yml:91]: Yes, indeed. I'll make sure of that.

**Review by alxhub — APPROVED** [review]

**Review by alxhub — APPROVED** [review]

**Comment by alxhub** [comment]
Caretaker: this is "green" in g3

**Comment by AndrewKushnir** [comment]
@JeanMeche there are a couple of merge conflicts with the `main` branch. Could you please rebase this PR and resolve conflicts when you get a chance?
